src/lib/map/analysis/MapSideNavigation.tsx

```diff
@@ -0,0 +1,159 @@
+import { useState } from 'react';
+import { SiteInformationProps } from './SemiTransparentHeader';
+import { ChevronRight } from '@veupathdb/coreui';
+import { Launch, LockOpen } from '@material-ui/icons';
+
+export type MapSideNavigationProps = {
+  children: React.ReactNode;
+  siteInformationProps: SiteInformationProps;
+};
+
+const menuBackground = 'rgba(255, 255, 255, 0.8)';
+
+const bottomLinkStyles: React.CSSProperties = {
+  // These are for formatting the links to the login
+  // and site URL.
+  display: 'flex',
+  justifyContent: 'flex-start',
+  alignItems: 'center',
+  fontSize: 15,
+  marginBottom: '1rem',
+};
+
+export function MapSideNavigation({
+  siteInformationProps,
+  children,
+}: MapSideNavigationProps) {
+  const [isExpanded, setIsExpanded] = useState<boolean>(true);
+
+  return (
+    <nav
+      style={{
+        // These styles define the look of the side navigation,
+        // including its width, height, and location on the
+        // page (which depends on the `isExpanded` piece of state.
+        background: menuBackground,
+        // height: 'calc(100% - 150px)',
+        height: '100%',
+        left: isExpanded ? 0 : -200,
+        minHeight: 125,
+        position: 'relative',
+        transition: 'left 0.1s ease',
+        width: 200,
```

Comment:
For maintainability, I think we should connect left's unexpanded offset to the container's width. Something could similarly be done to the button's width and its right offset. Perhaps some module-scoped vars that specify the widths, and the offsets can invert those values?

Reply:
Added in https://github.com/VEuPathDB/web-eda/pull/1671/commits/01f25de8d9f1177de8eb89ab0c50345f628330f2

Reply:
Yup, although I imagined them outside the component's scope!

Reply:
Gotcha! What's the use case you have in mind?

Reply:
No particular use case, I just imagine that we don't need to initialize those variables each time the component renders. Not sure if my reasoning is sound at all. And even it were sound, I know it makes no difference performance-wise at such a small scale. Just thinking out loud in a way!

Reply:
Makes sense! I think for now the variables make the CSS a bit easier to grokk, so I'm cool with them. I think we'll pick up this thread whenever we decide on what we're doing for CSS!

Reply:
When you resolve this, I would appreciate an explanation!

Reply:
(in a dev meeting follow-up topic or similar - don't spend hours typing)